core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -399,7 +402,7 @@ public void commit() {
     } catch (CommitStateUnknownException commitStateUnknownException) {
       throw commitStateUnknownException;
     } catch (RuntimeException e) {
-      Exceptions.suppressAndThrow(e, this::cleanAll);
+      cleanupIfPossible(e);
     }
```

Comment:
This doesn't seem right, we still want the suppress and throw behavior but the cleanup should be conditional on the strictCleanup and CleanableFailure